crates/tauri-bundler/src/bundle.rs

```diff
@@ -15,29 +16,33 @@ mod windows;
 
 use tauri_utils::{display_path, platform::Target as TargetPlatform};
 
+const BUNDLE_VAR_TOKEN: &[u8] = b"__TAURI_BUNDLE_TYPE_VAR_UNK";
 /// Patch a binary with bundle type information
 fn patch_binary(binary: &PathBuf, package_type: &PackageType) -> crate::Result<()> {
-  match package_type {
-    #[cfg(target_os = "linux")]
-    PackageType::AppImage | PackageType::Deb | PackageType::Rpm => {
-      log::info!(
-        "Patching binary {:?} for type {}",
-        binary,
-        package_type.short_name()
-      );
-      linux::patch_binary(binary, package_type)?;
-    }
-    PackageType::Nsis | PackageType::WindowsMsi => {
-      log::info!(
-        "Patching binary {:?} for type {}",
-        binary,
-        package_type.short_name()
-      );
-      windows::patch_binary(binary, package_type)?;
-    }
-    _ => (),
-  }
+  let mut file_data = std::fs::read(binary).expect("Could not read binary file.");
+
+  if let Some(bundle_var_index) = kmp::index_of(BUNDLE_VAR_TOKEN, &file_data) {
```

Comment:
Didn't look at it closely yet, but I think this would fail when patching the binary for the second bundle type right? (e.g `.msi` first and `__TAURI_BUNDLE_TYPE` becomes `__TAURI_BUNDLE_TYPE_VAR_MSI` already that when we do it for the NSIS bundle, we can no longer find `__TAURI_BUNDLE_TYPE_VAR_UNK`)

Reply:
do you remember if after the bundling is all done the binary in target/release/ will be a fresh one or the one from the last patch?

Reply:
I am still a bit concerned about this approach, like if we stopped at one of these steps (user pressed ctrl-c or we had an error somewhere), there's no way to recover/bundle again other than rebuilding the app's binary through cargo, and since we're matching any string in the binary, although unlikely, we risk replacing the wrong things

In my opinion, this method should not be used for the binary patching/main detection method

Reply:
in the future we should look into the "work on temp copy only" approach but that's not urgent

Reply:
So just to be clear, are we going to use this method for binary patching exclusively or do we want to use this as a fallback? (e.g. the find link section thing -> find the string and replace)